src/libraries/System.Security.Cryptography.X509Certificates/tests/DynamicChainTests.cs

```diff
@@ -450,6 +451,139 @@ public static void CustomTrustModeWithNoCustomTrustCerts()
+
+        [Fact]
+        public static void NameConstraintViolation_ExcludedTree_Dns()
+        {
+            SubjectAlternativeNameBuilder builder = new SubjectAlternativeNameBuilder();
+            builder.AddDnsName("www.example.com");
+
+            // excluded DNS name constraint for example.com.
+            string nameConstraints = "3012A110300E820C2E6578616D706C652E636F6D";
+
+            TestNameConstrainedChain(nameConstraints, builder, (bool result, X509Chain chain) => {
+                Assert.False(result, "chain.Build");
+                Assert.Equal(PlatformNameConstraints(X509ChainStatusFlags.HasExcludedNameConstraint), chain.AllStatusFlags());
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.OSX)] // macOS appears to just completely ignore min/max.
+        public static void NameConstraintViolation_PermittedTree_HasMin()
+        {
+            SubjectAlternativeNameBuilder builder = new SubjectAlternativeNameBuilder();
+            builder.AddDnsName("example.com");
+
+            // permitted DNS name constraint for example.com with a MIN of 9.
+            string nameConstraints = "3015A0133011820C2E6578616D706C652E636F6D800109";
+
+            TestNameConstrainedChain(nameConstraints, builder, (bool result, X509Chain chain) => {
+                Assert.False(result, "chain.Build");
+                Assert.Equal(PlatformNameConstraints(X509ChainStatusFlags.HasNotSupportedNameConstraint), chain.AllStatusFlags());
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.Windows)] // Windows seems to skip over nonsense GeneralNames.
+        public static void NameConstraintViolation_InvalidGeneralNames()
+        {
+            SubjectAlternativeNameBuilder builder = new SubjectAlternativeNameBuilder();
+            builder.AddEmailAddress("///");
+
+            // permitted RFC822 name constraint with GeneralName of ///.
+            string nameConstraints = "3009A007300581032F2F2F";
+
+            TestNameConstrainedChain(nameConstraints, builder, (bool result, X509Chain chain) => {
+                Assert.False(result, "chain.Build");
+                Assert.Equal(PlatformNameConstraints(X509ChainStatusFlags.InvalidNameConstraints), chain.AllStatusFlags());
+            });
+        }
+
+        private static X509ChainStatusFlags PlatformNameConstraints(X509ChainStatusFlags flags)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                const X509ChainStatusFlags AnyNameConstraintFlags =
+                    X509ChainStatusFlags.HasExcludedNameConstraint |
+                    X509ChainStatusFlags.HasNotDefinedNameConstraint |
+                    X509ChainStatusFlags.HasNotPermittedNameConstraint |
+                    X509ChainStatusFlags.HasNotSupportedNameConstraint |
+                    X509ChainStatusFlags.InvalidNameConstraints;
+
+                if ((flags & AnyNameConstraintFlags) > 0)
```

Comment:
```suggestion
                if ((flags & AnyNameConstraintFlags) != 0)
```

Seems more standard (not depending on the sign bit)